test/unit/test_keyring_raw_rsa.py

```diff
@@ -0,0 +1,311 @@
+
+
+def test_invalid_values_as_parameter():
+    with pytest.raises(Exception) as exc_info:
+        RawRSAKeyring(
+            key_namespace=_PROVIDER_ID,
+            key_name=_KEY_ID,
+            wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+            private_wrapping_key=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+        )
+    assert exc_info.errisinstance(TypeError)
+
+
+def test_public_and_private_key_not_provided():
+    with pytest.raises(TypeError) as exc_info:
+        RawRSAKeyring(
+            key_namespace=_PROVIDER_ID, key_name=_KEY_ID, wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1
+        )
+    assert exc_info.match("At least one of public key or private key must be provided.")
+
+
+def test_on_encrypt_when_data_encryption_key_given(patch_generate_data_key):
+    test_raw_rsa_keyring = RawRSAKeyring(
+        key_namespace=_PROVIDER_ID,
+        key_name=_KEY_ID,
+        wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+        private_wrapping_key=rsa.generate_private_key(
+            public_exponent=_PUBLIC_EXPONENT, key_size=_KEY_SIZE, backend=_BACKEND
+        ),
+    )
+
+    test_raw_rsa_keyring.on_encrypt(encryption_materials=get_encryption_materials_with_data_encryption_key())
+    # Check if keyring is generated
+    assert not patch_generate_data_key.called
+
+
+def test_keyring_trace_on_encrypt_when_data_encryption_key_given():
+    test_raw_rsa_keyring = RawRSAKeyring(
+        key_namespace=_PROVIDER_ID,
+        key_name=_KEY_ID,
+        wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+        private_wrapping_key=rsa.generate_private_key(
+            public_exponent=_PUBLIC_EXPONENT, key_size=_KEY_SIZE, backend=_BACKEND
+        ),
+    )
+
+    test = test_raw_rsa_keyring.on_encrypt(encryption_materials=get_encryption_materials_with_data_encryption_key())
+
+    for keyring_trace in test.keyring_trace:
+        if keyring_trace.wrapping_key.key_info == _KEY_ID:
+            # Check keyring trace does not contain KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY
+            assert KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY not in keyring_trace.flags
+
+
+def test_on_encrypt_when_data_encryption_key_not_given():
+    test_raw_rsa_keyring = RawRSAKeyring(
+        key_namespace=_PROVIDER_ID,
+        key_name=_KEY_ID,
+        wrapping_algorithm=WrappingAlgorithm.RSA_OAEP_SHA256_MGF1,
+        private_wrapping_key=rsa.generate_private_key(
+            public_exponent=_PUBLIC_EXPONENT, key_size=_KEY_SIZE, backend=_BACKEND
+        ),
+    )
+
+    original_number_of_encrypted_data_keys = len(
+        get_encryption_materials_without_data_encryption_key().encrypted_data_keys
+    )
+
+    test = test_raw_rsa_keyring.on_encrypt(encryption_materials=get_encryption_materials_without_data_encryption_key())
+
+    # Check if data key is generated
+    assert test.data_encryption_key and test.data_encryption_key is not None
```

Comment:
Ditto on falsy values.
```suggestion
    assert test.data_encryption_key is not None
```